### PR TITLE
EditPendingList use case (without screens)

### LINKED ISSUE #8: [Feature 8] Edit pending list
Allows group leaders to accept and reject users from their groups' pending lists.

### PR DESCRIPTION
**Pull request by SharonHsieh22**
EDIT: November 18, 2022

- EditPendingList and ViewPendingList use cases are basically finished
- Javadocs and method descriptions added (I formatted them like Aarya's)
- Added addGroup, removeFromApplications to User entity
- Added removeFromRequests, addMember to Group entity

Working on the pending list screen now.

*please ignore the the 11 commits I made, I was having merge conflicts but I finally learned how to solve those... if anyone has the same problem I might be able to help 

**also I accidently commited the files I pulled from main, which I don't think is necessary(?) I'm proabably going to close this pull request and make a new one later

### CODE REVIEW

**Review by aaryavatsa — DISMISSED** [review]
Added entity functions look good!

**Review by ipekakyol — DISMISSED** [review]
Both the changes and the newly added methods seem necessary and well done.

File: src/main/java/edit_pending_list_use_case/EditPendingListInteractor.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    public EditPendingListResponseModel addOrRemoveUser(EditPendingListRequestModel requestModel) {
+        Integer userID = requestModel.getUserID();
+        Integer groupID = requestModel.getGroupID();
+        boolean pendingStatus = requestModel.getPendingStatus();
+
+        if(!dsGateway.userExistsByID(userID)) {return presenter.prepareFailView("User does not exist.");}
+
+        // TODO: this shouldn't be possible, maybe make it an exception instead of presenting it to the user?
```
Comment (aaryavatsa) [inline src/main/java/edit_pending_list_use_case/EditPendingListInteractor.java]: You're so right about this actually.. in my use case I also checked for these weird border cases even though they shouldn't be possible. Would it be better to make them exceptions? Having these explicit checks makes my interactor code a lot longer.

**Review by aaryavatsa — DISMISSED** [review]
Added functions in the entity classes look good. Use case looks good too! Easy to follow along and very in-line with the clean architecture principles we've learned in class.

File: src/main/java/edit_pending_list_use_case/EditPendingListInteractor.java (hunk 1 above)
Comment (soheegoo) [reply src/main/java/edit_pending_list_use_case/EditPendingListInteractor.java]: Yes, I think having them as exceptions is a good idea.

**Pull request closed without merging**